CMakeLists.txt

```diff
@@ -249,14 +263,23 @@ else(MSVC)
     # -msse4.1 once marian can solely be compiled with intgemm ("onnxjs" will be removed in that case)
     set(INTRINSICS "-mssse3 -msimd128")
   else()
-    set(INTRINSICS "-msse4.1")
+    # Not assuming we have "-msse4.1 here"
```

Comment:
I get why you did this, but there are x86 performance implications.  Can you turn it back on for x86 builds?

Reply:
Sorry, but can you please expand the context and let me know what's the exact action item here. `BUILD_ARCH native` has a guard if SSE4.1 is found (after FindSSE) and then sets SSE4.1, the else-if is WASM this is the final else.

Reply:
If somebody isn't compiling for the exact host machine then the default assumption is that the target machine has at least `-msse4.1` support.  
I understand that ARM breaks this assumption.  
Would prefer it change to "if on x86, assume the machine has SSE4.1"

Reply:
[`3cf85f7` (#79)](https://github.com/browsermt/marian-dev/pull/79/commits/3cf85f703fa3bb54787631c13c8ff99aeb4e6dc0).

> if on x86 and not compiling native, assume the machine has SSE4.1

I have taken the liberty to alter this to "if SSE4.1 found irrespective of `-march=native`, add SSE4.1 support via flag". I find the objective weird, may I query which user/use-case are we worried about here?

Reply:
That's not quite what was asked for.  If somebody is building Marian without native, they are asking for a consistent x86 binary regardless of the host machine it was compiled on.  And currently that consistent binary is specced as SSE4.1. 

You've added conditions that check the host machine for SSE4.1 support and enable it if so in the non-native case.  

We can argue about whether that's the right assumption for x86, but that's not a change to be made in a PR about ARM support.  

The condition should be: if CMAKE_SYSTEM_PROCESSOR indicates x86 and native mode is not enabled, apply `-msse4.1`.  The second part (native mode is not enabled) already applies to this else clause, so all that remains is to put the x86 support back guarded by the target architecture.

Reply:
The first change has been reverted. [`e2069bf` (#79)](https://github.com/browsermt/marian-dev/pull/79/commits/e2069bfee29149ce38b06ba7b062daff454ac077) is in place.